rubix-spi/src/main/java/com/qubole/rubix/spi/CacheUtil.java

```diff
@@ -0,0 +1,327 @@
+/**
+ * Copyright (c) 2016. Qubole Inc
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License. See accompanying LICENSE file.
+ */
+package com.qubole.rubix.spi;
+
+import com.google.common.base.Charsets;
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import com.google.common.hash.HashCode;
+import com.google.common.hash.HashFunction;
+import com.google.common.hash.Hashing;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.Path;
+
+import java.io.File;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+
+public class CacheUtil
+{
+  private static final Log log = LogFactory.getLog(CacheUtil.class.getName());
+
+  private CacheUtil()
+  {
+  }
+
+  /**
+   * Create the necessary directories for caching file data.
+   *
+   * @param conf  The current Hadoop configuration.
+   */
+  public static void createCacheDirectories(Configuration conf)
+  {
+    final List<String> dirPrefixList = getDirPrefixList(conf);
+    final int maxDisks = CacheConfig.getCacheMaxDisks(conf);
+
+    for (String dirPrefix : dirPrefixList) {
+      for (int i = 0; i < maxDisks; ++i) {
+        final String cacheParentDir = dirPrefix + i;
+        log.debug("Checking for " + cacheParentDir);
+
+        if (exists(cacheParentDir)) {
```

Comment:
Throw exception if the cacheParentDir doesn't exist. There is no point of continuing as the directory where we are going to cache the files doesn't exist.

Reply:
Exception is now thrown when `cacheParentDir` does not exist